src/overlays/actors/ovl_En_Test3/z_en_test3.c

```diff
@@ -27,113 +203,1114 @@ const ActorInit En_Test3_InitVars = {
+    this->unk_D8D = ActorCutscene_GetAdditionalCutscene(this->player.actor.cutscene);
+    return true;
+}
+
+s32 func_80A3FE20(EnTest3* this, PlayState* play) {
+    struct_80A41828 sp2C;
+    ScheduleOutput schedule;
+
+    if (D_80A41D64 == 0) {
+        if (func_80A3E9DC(this, play)) {
+            sp2C.unk_1_0 = 2;
+            schedule.time0 = (u16)(gSaveContext.save.time - 0x3FFC);
+            schedule.time1 = (u16)(schedule.time0 + 0x3E8);
+            func_80A40098(this, play, &sp2C, &schedule);
+            D_80A41D64 = 1;
+            return false;
+        }
+    } else if (D_80A41D64 == 1) {
+        func_80A40230(this, play);
+    } else if (D_80A41D64 == 2) {
+        ActorCutscene_Stop(this->unk_D8D);
+        gSaveContext.save.weekEventReg[90] |= 2;
+        D_80A41D64 = 3;
+    }
+    return false;
+}
+
+s32 func_80A3FF10(EnTest3* this, PlayState* play, struct_80A41828* arg2, ScheduleOutput* arg3) {
+    static Vec3f D_80A418BC = { -420.0f, 210.0f, -162.0f };
+
+    if (gSaveContext.save.weekEventReg[51] & 0x40) {
+        D_80A41D68 = 2;
+        Math_Vec3f_Copy(&this->player.actor.world.pos, &D_80A418BC);
+        Math_Vec3f_Copy(&this->player.actor.home.pos, &D_80A418BC);
+
+        this->player.actor.home.rot.y = -0x2AAB;
+        this->player.actor.shape.rot.y = -0x2AAB;
+        this->player.currentYaw = -0x2AAB;
+        if (1) {} // macro?
+        return true;
+    } else {
+        func_80A3F15C(this, play, arg2);
+        this->unk_D8D = this->player.actor.cutscene;
+        if (play->roomCtx.currRoom.num == 2) {
+            this->unk_D8D = ActorCutscene_GetAdditionalCutscene(this->unk_D8D);
+        }
+        return true;
+    }
+}
+
+s32 func_80A3FFD0(EnTest3* this, PlayState* play2) {
+    PlayState* play = play2;
+
+    if (D_80A41D68 == 0) {
+        if (!Play_InCsMode(play) && (play->roomCtx.currRoom.num == 2)) {
+            D_80A41D68 = 1;
+        }
+    } else if (D_80A41D68 == 1) {
+        if (func_80A3E9DC(this, play)) {
+            D_80A41D68 = 2;
+        }
+    } else {
+        gSaveContext.save.weekEventReg[51] |= 0x40;
+        play->startPlayerCutscene(play, &this->player, 0x6E);
+    }
+    return false;
+}
+
+s32 func_80A40098(EnTest3* this, PlayState* play, struct_80A41828* arg2, ScheduleOutput* arg3) {
+    u16 sp26 = gSaveContext.save.time - 0x3FFC;
+    u16 temp;
+
+    func_80A3F15C(this, play, arg2);
+    this->unk_D7C = SubS_GetAdditionalPath(play, KAFEI_GET_PARAM1(&this->player.actor), ABS_ALT(arg2->unk_1_0) - 1);
+    if ((this->unk_D88 < 7) && (this->unk_D88 != 0) && (this->unk_D80 >= 0)) {
+        temp = sp26;
+    } else {
+        temp = arg3->time0;
+    }
+    if (arg3->time1 < temp) {
+        this->unk_DA8 = (temp - arg3->time1) + 0xFFFF;
+    } else {
+        this->unk_DA8 = arg3->time1 - temp;
+    }
+    this->unk_DB4 = sp26 - temp;
+    temp = this->unk_D7C->count - 2;
+    this->unk_DAC = this->unk_DA8 / temp;
```

Comment:
```suggestion
    temp2 = temp = this->unk_D7C->count - 2;
    this->unk_DAC = this->unk_DA8 / temp2;
```
Admittedly this is a hack, but if there is enough room on the stack for another `u16 temp2`, this makes naming things a bit easier. See `EnBaba_InitTimePath`